### PR TITLE
SDN-1521: oc-plumbing-related-refactoring

### PR DESCRIPTION
**Pull request by tssurya**
Though the change looks big, its not:

- I've refactored the folder structure under debug-scripts since in the image all scripts get copied into /usr/bin without retaining their folder structure. In future when we have more files, we can start using folders for arrangement.
- I've removed the `version` and `debug` files since they don't seem really necessary.
- I've moved some common functions into `common`.
- Added the integration script for oc plumbing `network-tools`

- SDN scripts were not working. Tidied them up and made them similar to OVN scripts.

TODOs after getting the client plumbing merged on the oc side:

- There will be upcoming PRs that add a basic set of instructions for developers when adding scripts to the repo so that there is _some_ sort of organization
- Refactor the `sdn_cluster_connectivity` and `sriov_connectivity` scripts.

/cc @rcarrillocruz @juanluisvaladas : PTAL.
@astoycos : I've also modified your script moving the pod creation portions to a more common place. You can also push a PR later on to change the rest of the script accordingly and remove the comment of the script from the `network-tools` call.

Advise to reviewers: Go commit by commit than going through all the files. It will be easier to reivew.

### CODE REVIEW

**Comment by tssurya** [comment]
/hold

until https://github.com/openshift/network-tools/pull/24 merges.

File: debug-scripts/common [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    if [ -z $resource_name ] || [ -z $(get_${type}_by_name $ns $resource_name) ]; then
+        # no existing resource is specified, so we need to create a new resource for the test.
+        resource_name="network-tools-debug"-${type}-$(get_random_name)
+        if [[ "$type" == "pod" ]]; then func=create_pod_on_node; else func=create_svc; fi
+        $func $resource_name $node
+        POD_NAMESPACE=$(oc get $type $resource_name -o jsonpath={.metadata.namespace})
+    else
+        POD_NAMESPACE=$ns
+    fi
+}
```
Comment (juanluisvaladas) [inline debug-scripts/common:130]: Per POSIX standard, every line must finish with a new line character in the end.

File: debug-scripts/sdn_pod_to_svc_connectivity [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        h) # display Help
+            help
+            exit;;
+    esac
+done
+
+src_name="${1}"
+dst_name="${2}"
+
+main
```
Comment (juanluisvaladas) [inline debug-scripts/sdn_pod_to_svc_connectivity:136]: Also missing the new line character in the end.

File: debug-scripts/sdn_node_connectivity [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+help()
+{
+    # Display Help
+    echo
+    echo "TODO"
+}
+
+main () {
+    help
```
Comment (juanluisvaladas) [inline debug-scripts/sdn_node_connectivity:53]: Do we always want to display help?

File: debug-scripts/sdn_pod_to_pod_connectivity [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    echo "oc adm network-tools -- sdn_pod_to_pod "" <dst-pod-namespace/dst-pod-name>"
+    echo "or"
+    echo "oc adm network-tools -- sdn_pod_to_pod <src-pod-namespace/src-pod-name>"
+    echo "or"
+    echo "podman run <IMAGE_ID> sdn_pod_to_pod"
+    echo
+}
+
+main () {
+    help
```
Comment (juanluisvaladas) [inline debug-scripts/sdn_pod_to_pod_connectivity:67]: Same.

File: debug-scripts/sdn_pod_to_svc_connectivity [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    echo "oc adm network-tools -- sdn_pod_to_svc "" <dst-svc-namespace/dst-svc-name>"
+    echo "or"
+    echo "oc adm network-tools -- sdn_pod_to_svc <src-pod-namespace/src-pod-name>"
+    echo "or"
+    echo "podman run <IMAGE_ID> sdn_pod_to_svc"
+    echo
+}
+
+main () {
+    help
```
Comment (juanluisvaladas) [inline debug-scripts/sdn_pod_to_svc_connectivity:121]: same

File: debug-scripts/ovn_pod_to_pod_connectivity [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    echo "oc adm network-tools -- ovn_pod_to_pod "" <dst-pod-namespace/dst-pod-name>"
+    echo "or"
+    echo "oc adm network-tools -- ovn_pod_to_pod <src-pod-namespace/src-pod-name>"
+    echo "or"
+    echo "podman run <IMAGE_ID> ovn_pod_to_pod"
+    echo
+}
+
+main () {
+    help
```
Comment (juanluisvaladas) [inline debug-scripts/ovn_pod_to_pod_connectivity:82]: same

File: debug-scripts/ovn_pod_to_svc_connectivity [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+    echo "oc adm network-tools -- ovn_pod_to_svc "" <dst-svc-namespace/dst-svc-name>"
+    echo "or"
+    echo "oc adm network-tools -- ovn_pod_to_svc <src-pod-namespace/src-pod-name>"
+    echo "or"
+    echo "podman run <IMAGE_ID> ovn_pod_to_svc"
+    echo
+}
+
+main () {
+    help
```
Comment (juanluisvaladas) [inline debug-scripts/ovn_pod_to_svc_connectivity:82]: same

File: debug-scripts/network-tools [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    ovn_pod_to_svc_connectivity "$global_namespace"/"$client" "$global_namespace"/"$server"
+    # ovn_ipsec_connectivity
+elif [[ "$network_plugin" == "OpenShiftSDN" ]] ; then
+    # run scripts
+    sdn_pod_to_pod_connectivity "$global_namespace"/"$client" "$global_namespace"/"$server"
+    sdn_pod_to_svc_connectivity "$global_namespace"/"$client" "$global_namespace"/"$server"
+    # sdn_node_connectivity
+else
+    echo "Unable to debug cluster networking. Only OpenShiftSDN and OVNKubernetes plugins are supported"
+fi
```
Comment (juanluisvaladas) [inline debug-scripts/network-tools:31]: again the newline

**Review by juanluisvaladas — CHANGES_REQUESTED** [review]
Seems pretty good, I just have a couple nitpicks.

File: debug-scripts/sdn_node_connectivity (hunk 3 above)
Comment (tssurya) [reply debug-scripts/sdn_node_connectivity:53]: yea agreed not needed.

**Comment by tssurya** [comment]
> Seems pretty good, I just have a couple nitpicks.

thanks Juan for the review. Have addressed the nits.

**Pull request merged**